src/uct/ib/dc/dc_mlx5_ep.c

```diff
@@ -1378,24 +1378,27 @@ void uct_dc_mlx5_ep_handle_failure(uct_dc_mlx5_ep_t *ep, void *arg,
     ucs_arbiter_group_t *group;
     uint8_t pool_index;
 
+    ucs_debug("handle failure iface: %p, dci[%d] qpn 0x%x, status: %s", iface,
+              dci_index, txwq->super.qp_num, ucs_status_string(ep_status));
+
     ucs_assert(!uct_dc_mlx5_iface_is_dci_rand(iface));
 
     uct_dc_mlx5_update_tx_res(iface, txwq, txqp, pi);
     uct_rc_txqp_purge_outstanding(&iface->super.super, txqp, ep_status, pi, 0);
 
     ucs_assert(ep->dci != UCT_DC_MLX5_EP_NO_DCI);
```

Comment:
btw, what prevents this assert to fail, when we get several CQE with error in a row?

Reply:
never mind, will be handled by `uct_dc_mlx5_dci_handle_failure`